sos/plugins/postgresql.py

```diff
@@ -107,29 +113,29 @@ def postproc(self):
 
 class RedHatPostgreSQL(PostgreSQL, RedHatPlugin):
 
+    packages = ('postgresql', 'rh-postgresql95-postgresql-server')
+
     def setup(self):
+        if self.is_installed('postgresql'):
+            self.pghome = self.get_option("pghome")
+        else:  # SCL postgreSQL is installed
```

Comment:
The patch looks good to me. Just one question, isn't a good idea to check if  rh-postgresql95-postgresql-server is installed as you did in line 119?

Reply:
This class can be enabled either by:

- presence of either `postgresql` or `rh-postgresql95-postgresql-server` package (plugin/class is enabled by default when a package from `packages` is present, or when a file from `files` is present) - see https://github.com/sosreport/sos/pull/1118/files#diff-93be9976582cadc41ed1cec376205196R116 and https://github.com/sosreport/sos/blob/master/sos/plugins/__init__.py#L99-L102 . In this case, and if `postgresql` isnt installed, we can deduce ` rh-postgresql95-postgresql-server` is installed.
- or this class can be enabled manually, e.g. via cmdline option `-o postgres`. Then potentially neither package can be present. In that case, we should rather take default postgres data as the default option. So we should rather revert the test (use `scl` commands only when `scl` installed, otherwise default to normal `postgres`).

The difference is imho small, leaving on @bmr-cymru opinion if it makes sense to improve it that way.

Reply:
Eh, this does not look right to me.

Having separate `pghome` and `sclpghome` plugin options is a bit of an ugly hack, and not very nice for users to try to understand.

I think it would be better to revive the `SCLPluign` mixin concept from PR #900 than to attempt to solve this in an ad-hoc and plugin specific fashion.